third-party/gasnet/Makefile

```diff
@@ -91,8 +91,9 @@ GASNET_CFLAGS += $(SHARED_LIB_CFLAGS)
 MPI_CC += $(SHARED_LIB_CFLAGS)
 endif
 
-# Disable hwloc until https://github.com/chapel-lang/chapel/issues/18050
-CHPL_GASNET_CFG_OPTIONS += --disable-hwloc
+ifeq ($(CHPL_MAKE_HWLOC),bundled)
+CHPL_GASNET_CFG_OPTIONS += --with-hwloc-home=$(HWLOC_INSTALL_DIR)
+endif
```

Comment:
I see in the OP that you tested this with CHPL_HWLOC=system, but nothing in this diff looks like it will guaranteed that Chapel and GASNet use the same hwloc install. Can you add the following so that this is guaranteed?

```
ifeq ($(CHPL_MAKE_HWLOC),system)
  CHPL_GASNET_CFG_OPTIONS += --with-hwloc-home=$(shell $(CHPL_MAKE_PYTHON) $(CHPL_MAKE_HOME)/util/chplenv/chpl_hwloc.py --prefix)
endif
```

Reply:
Good point, I was just assuming that each would reliably find the same system hwloc install in the `$PATH`. However I'll make the connection explicit as you suggest.